exercism/diamond/diamond.js

```diff
@@ -0,0 +1,13 @@
+var Diamond = function () {}
```

Comment:
This one was super tricky for my pair and I. I'm not sure if the remaining code wasn't pushed up, but I'll add our solution so you can take a look at it. It's not the nicest, but I think we got it to work:

function diamond(){
  var letters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'.split('');

  this.makeDiamond = function(letter) {

    var count = letters.indexOf(letter);

    var result = [];

    for(var i = 0; i <= count; i++) {
      if(i === 0){
        result.push(" ".repeat(count - i) + letters[i] + " ".repeat(count - i));
      } else {
        result.push(" ".repeat(count - i) + letters[i] + " ".repeat(i * 2 - 1) + letters[i] + " ".repeat(count-i));
      }
    }

    var reverse = result.slice(0, count).reverse();

    if(count === 0) {
      return "A" + "\n";
    } else {
      return result.join("\n") + "\n" + reverse.join("\n") + "\n";
    }
  }
}